lib/rubygems/command.rb

```diff
@@ -323,6 +323,10 @@ def invoke_with_build_args(args, build_args)
     elsif @when_invoked
       @when_invoked.call options
     else
+      if Gem.paths.auto_user_install && !options[:install_dir] && !options[:user_install]
+        self.ui.say "Defaulting to user installation because default GEM_HOME (#{Gem.default_dir}) is not writable."
```

Comment:
Could this mislead users into thinking they have set `GEM_HOME` (the env variable)? Maybe

```suggestion
        self.ui.say "Defaulting to user installation because default installation directory (#{Gem.default_dir}) is not writable."
```
?